sled-agent/src/services.rs

```diff
@@ -1093,24 +1103,21 @@ impl ServiceManager {
                             smfh.refresh()?;
                         }
                         ServiceType::Dendrite { .. } => {
-                            smfh.setprop(
-                                "config/address",
-                                &format!("[{}]:{}", address, DENDRITE_PORT),
-                            )?;
+                            smfh.delpropvalue("config/address", "*")?;
+                            for address in &request.addresses {
+                                smfh.addpropvalue(
+                                    "config/address",
+                                    &format!("[{}]:{}", address, DENDRITE_PORT),
+                                )?;
+                            }
                             smfh.refresh()?;
                             // TODO: For this restart to be optional, Dendrite must
                             // implement a non-default "refresh" method.
                             smfh.restart()?;
```

Comment:
I saw that there was ongoing work here - presumably, we'll be able to refresh Dendrite without a restart soon too, right?

Reply:
Yes.  That's currently in review.